### ISSUE TITLE
Additional logging for MLFlow

### ISSUE THREAD

**Issue by tonyreina** [ultralytics/ultralytics#18365]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I'm able to train and pass the training logs to my local MLFlow server. However, there are other artifacts that I would like to log (at least manually if not automatically). The final model does not seem to be logging although when I looked at the Ultralytics code on [GitHub it does appear to try to do this (I think?)](https://github.com/ultralytics/ultralytics/blob/da9d8730f9f8a99fe6cd980fc927fd1005c15676/ultralytics/utils/callbacks/mlflow.py#L112)

If I simply do `mlflow.log_artifact("yolo.pt")` after training, then a brand new run is created. How can I get the run ID of the Ultralytics `model.train` routine?

```
import os
from ultralytics import YOLO
import mlflow

mlflow.set_tracking_uri("http://localhost:5000")

data_dir = os.path.join(os.getcwd(), "data")
model_name = "yolo-one2one"
project_name = "yolo-one2one-model"

# Create a YAML configuration for YOLO training
yaml_content = f"""
train: {data_dir}/train  # Path to training images and labels
val: {data_dir}/val      # Path to validation images and labels
test: {data_dir}/test    # Path to test images and labels

nc: 1

names:
  - plane
"""

yaml_file = "data.yaml"
with open(yaml_file, "w") as f:
    f.write(yaml_content)

# Load YOLO base model
model_base = "yolo11n.pt"
model = YOLO(model_base)

# Check for previous training
resume_path = os.path.join("runs", "detect", model_name, "weights", "last.pt")
resume = os.path.isfile(resume_path)

# Train the model
results = model.train(
    data=yaml_file,
    epochs=10,
    imgsz=640,
    name=model_name,
    project=project_name,
    single_cls=True,
    mosaic=0.2,
    translate=0.1,
    scale=0.0,
    seed=816,
    resume=resume,
)
```

**Comment by UltralyticsAssistant** [comment]
👋 Hello @tonyreina, thank you for your interest in Ultralytics 🚀! We recommend checking out the [Docs](https://docs.ultralytics.com) for additional support, as it offers comprehensive resources, including [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples.

It looks like your issue is about adding custom logging with MLFlow. To assist further, if this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/). This would help us understand and debug your code more effectively. If this is a usage ❓ Question, it's helpful to include as much detail as possible about your setup, such as the versions of Ultralytics, Python, and MLFlow you're using, along with any training logs or specific error messages.

### Community and Resources
Here are a few resources that might help you connect with the community or find quick answers:
- **Join our Discord for live discussions** 🎧: [Ultralytics Discord](https://discord.com/invite/ultralytics)
- **Explore in-depth topics with Discourse** 📝: [Ultralytics Discourse](https://community.ultralytics.com)
- **Discuss or share ideas on Reddit**: [r/Ultralytics](https://reddit.com/r/Ultralytics)

### Upgrade and Environment Verification
Ensure you're using the latest release of Ultralytics for the most up-to-date features and improvements. Run the following command to upgrade your setup:

```bash
pip install -U ultralytics
```

Make sure your environment satisfies all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml), including [Python>=3.8](https://www.python.org/) and [PyTorch>=1.8](https://pytorch.org/get-started/locally/).

Additionally, you can use the following verified environments to ensure compatibility:
- Notebooks with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- See ML-focused options such as [Google Cloud](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/), [AWS](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/), or [Docker](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a> to streamline your workflows!

### Continuous Integration (CI) Status
You can check the CI status here: [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule). <a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

An Ultralytics engineer will review your question and provide guidance soon. In the meantime, thank you for your patience 😊.

**Comment by Y-T-G** [comment]
https://github.com/ultralytics/ultralytics/blob/da9d8730f9f8a99fe6cd980fc927fd1005c15676/ultralytics%2Futils%2Fcallbacks%2Fmlflow.py#L116

You can set this environment variable to True so that it keeps the run alive.

**Comment by tonyreina** [comment]
Thanks!

Why doesn't the save model code get triggered?  Is there another environment variable that I need to define?

https://github.com/ultralytics/ultralytics/blob/da9d8730f9f8a99fe6cd980fc927fd1005c15676/ultralytics%2Futils%2Fcallbacks%2Fmlflow.py#L112

**Comment by tonyreina** [comment]
Oddly I can set tags and log params, but when I try to log artifacts (such as the model), then nothing gets added to MLFlow. Might be on my MLFlow side.

**Comment by Y-T-G** [comment]
Does it work if you use `log_model()` instead?

https://mlflow.org/docs/latest/python_api/mlflow.pytorch.html#mlflow.pytorch.log_model

**Comment by tonyreina** [comment]
I'll try. I think I've got my docker compose wrong on MLFlow. So the runs are logged, but the artifacts are not. So not a Ultralytics issue, but a tony issue.

**Comment by glenn-jocher** [comment]
Thank you for clarifying! It does sound like an MLFlow configuration issue. If you need further assistance with Ultralytics-specific functionality, feel free to reach out.
